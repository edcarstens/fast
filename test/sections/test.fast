======================================================
TEST: subsections
------------------------------------------------------
//.top = Section('top')
//.+top.a
Section A
------------------------------------------------------
  //.top.a > 'a1'   ## print subsection a1
AA----------------------------------------------------
  //.top.a.a2 > ''  ## another way to do it
//.-top.a
//.+top.b
Section B
------------------------------------------------------
parent of top.a is '@top.a.parent.section@'
parent of top.a.a1 is '@@a1.parent.section@@'
BB----------------------------------------------------
//.top > 'a'
------------------------------------------------------
//.-top.b
//.+top.a.a1
Section A1
//.-top.a.a1
//.a1 = top.a.a1
//.+top.a.a2
Section A2
//.-top.a.a2
//.### Peek at current text in Section A ###
@top.a.getText()@
------------------------------------------------------
//.### Peek at current text in Section B ###
@top.b.getText()@
------------------------------------------------------
//.top > 'a'
------------------------------------------------------
//.top.b > ''
------------------------------------------------------

======================================================
TEST: fast.seal, fast.unseal
------------------------------------------------------
//.+top
//.top > 'b'   ## sealed
//.-top


XX----------------------------------------------------
@top.getText()@

//.fast.unseal()
//.+top
//.top > 'b'   ## this unseals Section B
//.-top
//.## (top still has some sealed sections)
//.fast.seal()

YY----------------------------------------------------
@top.getText()@


ZZ----------------------------------------------------
//.top > ''   ## unseals output of Section top

ZZ----------------------------------------------------
//.### But Section top itself is still sealed
@top.getText()@


======================================================
TEST: Section.write, fast.writeFile
------------------------------------------------------
//.#fast.seal()
//.#top.write('top.txt')
//.top('top.txt').write()
======================================================
TEST: Write to file (testfile)
------------------------------------------------------
//.top = Section('top')
//.+top.aa
section aa
//.-top.aa
//.+top.testfile('testfile.txt')
//.top > 'aa'
//.-top.testfile.write()
