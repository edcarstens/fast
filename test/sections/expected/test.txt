======================================================
TEST: subsections
------------------------------------------------------
Section A
------------------------------------------------------
  //.fast.sections['top a a1'].processFinal()
  //.fast.includeSection('top a a1')
AA----------------------------------------------------
  //.fast.sections['top a a2'].processFinal()
  //.fast.includeSection('top a a2')

------------------------------------------------------
Section B
------------------------------------------------------
parent of top.a is 'top'
parent of top.a.a1 is '@a1.parent.section@'
BB----------------------------------------------------
//.fast.sections['top a'].processFinal()
//.fast.includeSection('top a')
------------------------------------------------------

------------------------------------------------------
Section A
------------------------------------------------------
  Section A1
AA----------------------------------------------------
  Section A2
------------------------------------------------------
Section B
------------------------------------------------------
parent of top.a is 'top'
parent of top.a.a1 is 'top a'
BB----------------------------------------------------
Section A
------------------------------------------------------
  Section A1
AA----------------------------------------------------
  Section A2
------------------------------------------------------
------------------------------------------------------

======================================================
TEST: fast.seal, fast.unseal
------------------------------------------------------


XX----------------------------------------------------
//.fast.includeSection('top b')



YY----------------------------------------------------
//.fast.includeSection('top b')
Section B
------------------------------------------------------
parent of top.a is 'top'
parent of top.a.a1 is 'top a'
BB----------------------------------------------------
Section A
------------------------------------------------------
  Section A1
AA----------------------------------------------------
  Section A2
------------------------------------------------------



ZZ----------------------------------------------------
Section B
------------------------------------------------------
parent of top.a is 'top'
parent of top.a.a1 is 'top a'
BB----------------------------------------------------
Section A
------------------------------------------------------
  Section A1
AA----------------------------------------------------
  Section A2
------------------------------------------------------
Section B
------------------------------------------------------
parent of top.a is 'top'
parent of top.a.a1 is 'top a'
BB----------------------------------------------------
Section A
------------------------------------------------------
  Section A1
AA----------------------------------------------------
  Section A2
------------------------------------------------------

ZZ----------------------------------------------------
//.fast.includeSection('top b')
Section B
------------------------------------------------------
parent of top.a is 'top'
parent of top.a.a1 is 'top a'
BB----------------------------------------------------
Section A
------------------------------------------------------
  Section A1
AA----------------------------------------------------
  Section A2
------------------------------------------------------



======================================================
TEST: Section.write, fast.writeFile
------------------------------------------------------
SectionBase.write: Info - writing section (top) to file "top.txt"
======================================================
TEST: Write to file (testfile)
------------------------------------------------------
SectionBase.write: Info - writing section (top testfile) to file "testfile.txt"
